examples/gno.land/p/demo/paginate/paginate.gno

```diff
@@ -12,22 +19,116 @@ type renderer interface {
 // TODO this function belongs somewhere convenient.
 // escape a URL for markdown.
 func escMDURL(s string) string {
-	// XXX implement
+	schars := map[string]string{
+		"%":  "%25",
+		" ":  "%20",
+		"!":  "%21",
+		"\"": "%22",
+		"$":  "%24",
+		"&":  "%26",
+		"'":  "%27",
+		"(":  "%28",
+		")":  "%29",
+		"*":  "%2A",
+		"+":  "%2B",
+		",":  "%2C",
+		";":  "%3B",
+		"<":  "%3C",
+		">":  "%3E",
+		"?":  "%3F",
+		"@":  "%40",
+		"[":  "%5B",
+		"\\": "%5C",
+		"]":  "%5D",
+		"^":  "%5E",
+		"_":  "%5F",
+		"`":  "%60",
+		"{":  "%7B",
+		"|":  "%7C",
+		"}":  "%7D",
+		"~":  "%7E",
+		"\n": "%0A",
+	}
+	for k, v := range schars {
+		s = strings.Replace(s, k, v, -1)
+	}
+	return s
 }
 
-var itoa = strconv.Itoa
+func navigation(numPages int, current int, reverse bool, pageFn pageFn) string {
+	str := ""
+
+	i := 1
+	for ; i <= PAGINATE_GAP && i < current; i++ {
+		pageurl := pageFn(i, numPages, reverse)
+		str += "[" + strconv.Itoa(i) + "](" +
+			escMDURL(pageurl) + ") "
```

Comment:
```suggestion
		str += md.Link(strconv.Itoa(i),  pageurl)
```

if you go with a new `md` library with `md.EscapeURL(s string)` and `md.Link(title, url string)`